src/report.ts

```diff
@@ -30,27 +34,55 @@ async function pollReport(
     throw new Error(`"project-name" must not contain spaces or special characters except [/-_]`);
   }
 
+  return initAndPollReportGeneric(initReport, getReport, options);
+}
+
+async function initAndPollScmReport(
+  options: ScmUploadReportOptions,
+): Promise<Result<AnalysisFailedResponse | ReportResult, GetAnalysisErrorCodes>> {
+  const projectId = options.projectId?.trim();
+  if (!projectId || projectId.length === 0) {
+    throw new Error('"project-id" must be provided for "report"');
+  }
+  if (!uuid.validate(projectId)) {
+    throw new Error('"project-id" must be a valid UUID');
+  }
+
+  const commitId = options.commitId?.trim();
+  if (!commitId || commitId.length === 0) {
```

Comment:
AFAIK long is preferred but short should also work.
If you're asking because of validation, there's not much we can do client-side because different SCMs can have different hash lengths, so I figured it's best to do a basic check and rely on SBS to do proper validation when it tries to clone the repo.